app/submitter/convert_payload_0_0_3.py

```diff
@@ -118,21 +118,12 @@ def add_relationship_answers(
         relationships_block_id=relationships_block["id"],
         unrelated_block_id=unrelated_block_id,
         unrelated_answer_id=unrelated_answer_id,
+        unrelated_no_answer_value=unrelated_no_answer_value,
     )
 
-    relationship_answers = []
     for location in relationship_router.path:
-        if location.to_list_item_id:
-            if relationship_answer := relationship_store.get_relationship(
-                location.list_item_id, location.to_list_item_id
-            ):
-                relationship_answers.append(relationship_answer)
-        else:
+        if location.block_id == unrelated_block_id:
             if unrelated_answer := answer_store.get_answer(
                 unrelated_answer_id, list_item_id=location.list_item_id
             ):
```

Comment:
These could be combined. Not sure if how walrus operator plays nice when it is not the first expression.

Reply:
I think I avoided this because the walrus expression needed to be wrapped in brackets, and thought it would be less readable with the nested brackets for the `get_answer` call.

I've tried this and it actually looks more readable than I thought it would, once it's formatted with black it ends up looking like this:

```python
        if location.block_id == unrelated_block_id and (
            unrelated_answer := answer_store.get_answer(
                unrelated_answer_id, list_item_id=location.list_item_id
            )
        ):
            answers_payload.add_or_update(unrelated_answer)
```

What do you think?

Reply:
I think that is fine and I prefer it over the two ifs.

Reply:
Yes, I ran lint and all lint checks passed 🙂

Reply:
I've just pushed this change.